app/assets/javascripts/components/ViewAll.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * # ViewAll
+ *
+ * Dough component to display an initial set of items,
+ * and show the remaining items upon a trigger being invoked.
+ *
+ * @module ViewAll
+ * @returns {class} ViewAll
+ */
+
+define(['jquery', 'DoughBaseComponent'],
+       function($, DoughBaseComponent) {
+  'use strict';
+
+  var ViewAllProto,
+      defaultConfig = {
+        initialNumberOfItemsToDisplay: 5,
+        itemSelector: '[data-dough-view-all-item]',
+        triggerSelector: '[data-dough-view-all-trigger]'
+      };
+
+  /**
+   * @constructor
+   * @extends {DoughBaseComponent}
+   * @param {HTMLElement} $el    Element with dough-component on it
+   * @param {Object}             config Hash of config options
+   */
+  function ViewAll($el, config) {
+    ViewAll.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   */
+  DoughBaseComponent.extend(ViewAll);
+  ViewAll.componentName = 'ViewAll';
+  ViewAllProto = ViewAll.prototype;
+
+  /**
+   * Init function
+   *
+   * Set up listeners and accept promise
+   *
+   * @param {Object} initialised Promise passed from eventsWithPromises (RSVP Promise).
+   * @returns {MultiTableFilter}
+   */
+  ViewAllProto.init = function(initialised) {
+    this._hideAdditionalItems();
+    this._displayTrigger();
+    this._bindTriggerEvent();
+    this._initialisedSuccess(initialised);
+    return this;
+  };
+
+  /**
+   * _hideAdditionalItems
+   *
+   * If there are more items than the initial number to display, it
+   * applies a style to hide them.
+   *
+   * @private
+   */
+  ViewAllProto._hideAdditionalItems = function() {
+    var count = this.config.initialNumberOfItemsToDisplay - 1;
+    var selector = this.config.itemSelector + ':gt('+ count +')';
+    this.$find(selector).addClass('view-all__item--hidden');
+  };
+
+  /**
+   * _displayTrigger
+   *
+   * If additional items are still to be displayed, removes the
+   * class which hides the trigger
+   *
+   * @private
+   */
+  ViewAllProto._displayTrigger = function() {
+    if(this._moreItemsToDisplay()) {
+      this.$trigger().removeClass('view-all__trigger--hidden');
+    }
+  }
+
+  /**
+   * _bindTriggerEvent
+   *
+   * Attaches a click event to each trigger to display the
+   * remaining additional items.
+   *
+   * @private
+   */
+  ViewAllProto._bindTriggerEvent = function() {
+    var self = this;
+    this.$trigger().on('click', function(e) {
```

Comment:
Personally, I prefer to move EventHandlers out into their own private functions, so this line would become

this.$trigger.on('click', $.proxy( this._triggerClickHandler, this ) );

This will then allow you to clean up your EventHandlers to reduce memory leaks, you can't do that properly using the short-hand version because there's no real signature.

The other benefit is that you can loose the var self and then change the overall function name from "_bindTriggerEvent" to "_bindEvents" then each handler denotes what is happening.